src/components/view/View.spec.tsx

```diff
@@ -63,474 +66,474 @@ describe('View', () => {
       typeId: type,
       diagramMakerData: {
         position: { x, y },
-        size: { width: 100, height: 100 }
+        size: { width: 100, height: 100 },
       },
-      consumerData: undefined
-    }
+      consumerData: undefined,
+    },
   });
 
   const generateEdge = (edgeKey: string, edgeSrc: string, edgeDest: string) => ({
     [edgeKey]: {
       dest: edgeDest,
       diagramMakerData: {},
       id: edgeKey,
-      src: edgeSrc
-    }
+      src: edgeSrc,
+    },
   });
 
   const generateContextMenu = (
-    targetType: DiagramMakerComponentsType, targetId?: string, position: Position = { x: 0, y:0 }
+    targetType: DiagramMakerComponentsType,
+    targetId?: string,
+    position: Position = { x: 0, y: 0 },
   ) => ({
     position,
     targetId,
-    targetType
+    targetType,
   });
 
   const generatePotentialEdge = (edgeSrc: string, x = 0, y = 0) => ({
     position: { x, y },
-    src: edgeSrc
+    src: edgeSrc,
   });
 
   const generatePotentialNode = (id: string, x = 0, y = 0, width = 150, height = 100) => ({
     position: { x, y },
     size: { height, width },
-    typeId: id
+    typeId: id,
   });
 
   const generatePanel = (panelKey: string, x = 0, y = 0) => ({
     [panelKey]: {
       id: panelKey,
       position: {
         x,
-        y
+        y,
       },
       size: {
         height: 300,
-        width: 300
-      }
-    }
+        width: 300,
+      },
+    },
   });
 
   describe('nodes with edge', () => {
-    const mockNodes = Object.assign({}, generateNode('node-1'), generateNode('node-2', 100, 200));
-    const mockEdges = Object.assign({}, generateEdge('edge-1', 'node-1', 'node-2'));
+    const mockNodes = { ...generateNode('node-1'), ...generateNode('node-2', 100, 200) };
+    const mockEdges = { ...generateEdge('edge-1', 'node-1', 'node-2') };
```

Comment:
i don't understand why this must be spread?

Reply:
Thats because generateNode returns an object with a single key i.e. nodeId and the value is the actual node object, hence must be spread again into the final nodes object containing all nodes. This is what generateNode returns:

```
{
  'node-1': { 'id': 'node-1', 'position': { 'x': 10, 'y': 10 }, 'size': { 'width': 100, 'height': 100 } }
}
```